xclim/sdba/adjustment.py

```diff
@@ -318,6 +334,168 @@ def adjust(
+                cls._check_inputs(ref, hist, group=kwargs["group"])
+
+            # I don't understand the point of this line, this is done above already
+            # hist = convert_units_to(hist, ref)
+        else:
+            train_units = ""
+
+        ds, params = cls._train(ref, hist, **kwargs)
+        obj = cls(
+            _trained=True,
+            hist_calendar=get_calendar(hist),
+            train_units=train_units,
+            **params,
+        )
+        obj.set_dataset(ds)
+        return obj
+
+    def adjust(self, sim: DataArray, *args, **kwargs):
+        r"""Return bias-adjusted data.
+
+        Refer to the class documentation for the algorithm details.
+
+        Parameters
+        ----------
+        sim : xr.Dataset
+            Time series to be bias-adjusted, usually a model output.
+        args : xr.Dataset | xr.DataArray
+            Other Datasets/DataArrays needed for the adjustment (usually none).
+        \*\*kwargs
+            Algorithm-specific keyword arguments, see class doc.
+        """
+        skip_checks = kwargs.pop("skip_input_checks", False)
+        if not skip_checks:
+            (sim, *args), _ = self._harmonize_units(sim, *args, target=self.train_units)
+
+            if "group" in self:
+                self._check_inputs(sim, *args, group=self.group)
+
+            # sim = convert_units_to(sim, self.train_units)
+        scen = self._adjust(sim, *args, **kwargs)
+
+        # Keep attrs
+        scen.attrs.update(sim.attrs)
+        for name, crd in sim.coords.items():
+            if name in scen.coords:
+                scen[name].attrs.update(crd.attrs)
+        params = gen_call_string("", **kwargs)[1:-1]  # indexing to remove added ( )
+        infostr = f"{str(self)}.adjust(sim, {params})"
+        scen.attrs["history"] = update_history(f"Bias-adjusted with {infostr}", sim)
+        scen.attrs["bias_adjustment"] = infostr
+        return scen
+
+    def set_dataset(self, ds: xr.Dataset):
+        """Store an xarray dataset in the `ds` attribute.
+
+        Useful with custom object initialization or if some external processing was performed.
+        """
+        super().set_dataset(ds)
+        self.ds.attrs["adj_params"] = str(self)
+
+    @classmethod
+    def _train(cls, ref: DataArray, hist: DataArray, *kwargs):
+        raise NotImplementedError()
+
+    def _adjust(self, sim, **kwargs):
+        raise NotImplementedError()
+
+
+class MultivariateAdjust(BaseAdjustment):
+    """Adjustment with no intermediate trained object.
+
+    Children classes should implement a `_adjust` classmethod taking as input the three Datasets
+    and returning the scen dataset/array.
+    """
+
+    @classmethod
+    def adjust(
+        cls,
+        ref: xr.Dataset,
+        hist: xr.Dataset,
+        sim: xr.Dataset | None = None,
+        **kwargs,
+    ) -> xr.Dataset:
+        r"""Return bias-adjusted data. Refer to the class documentation for the algorithm details.
```

Comment:
`sim` to be optional: I think it makes sense in a purely `adjust` Class to allow this and use directly `hist`  which is also in the arguments. And @LamAdr has the OTC function that performs an adjustment specifically on `hist`, not some other dataset (for another dataset, this is dOTC). 

In principle, I have the intution that OTC should in fact be generalized to accept a different `sim` than `hist`, say you train on a member-1, but you adjust member-2, in principles those two datasets have the same statistics, so it could be interesting. But it's not described in the literature I believe, it would required experiments.  For now, we can specifically ask for OTC that we use `sim == hist`